flycheck.el

```diff
@@ -8103,6 +8103,25 @@ for more information about the custom directories."
                                                  "--print-config" "."))))
     (eq exitcode 0)))
 
+(defun flycheck-eslint--find-working-directory (_checker)
+  "Look for a working directory to run CHECKER in.
+This will either be the directory that contains `.eslintrc' in some of
+supports configuration files formats, if no such file is found in the directory
+hierarchy, searches `node_modules' directory to detect root of project."
+  (let* ((config-regexp (concat "\\`"
+                                (regexp-quote ".eslintrc")
+                                "\\(\\.\\(js\\|ya?ml\\|json\\)\\)?\\'")))
+    (or
+     (and
+      buffer-file-name
+      (or (locate-dominating-file
+           default-directory
+           (lambda (directory)
+             (> (length (directory-files directory nil config-regexp 0)))))
+          (locate-dominating-file buffer-file-name "node_modules")
+          (file-name-directory buffer-file-name)))
```

Comment:
I'm not sure this line is needed — isn't default-directory set to that by default?

Reply:
No really, no for eslint

Reply:
I think it's always the case that opening a file in Emacs sets default directory to the containing folder, so I wouldn't worry about removing this line :)